src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,878 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
-            // calling Beta.B.Write to fail.
-            loader.AddDependencyLocation(_testFixture.Gamma.Path);
-            loader.AddDependencyLocation(_testFixture.Beta.Path);
-            Assembly beta = loader.LoadFromPath(_testFixture.Beta.Path);
-
-            var b = beta.CreateInstance("Beta.B")!;
-            var writeMethod = b.GetType().GetMethod("Write")!;
-            var exception = Assert.Throws<TargetInvocationException>(
-                () => writeMethod.Invoke(b, new object[] { sb, "Test B" }));
-            Assert.IsAssignableFrom<FileNotFoundException>(exception.InnerException);
-
-            var actual = sb.ToString();
-            Assert.Equal(@"", actual);
+            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                StringBuilder sb = new StringBuilder();
+
+                // We don't pass Alpha's path to AddDependencyLocation here, and therefore expect
+                // calling Beta.B.Write to fail.
+                loader.AddDependencyLocation(testFixture.Gamma.Path);
+                loader.AddDependencyLocation(testFixture.Beta.Path);
+                Assembly beta = loader.LoadFromPath(testFixture.Beta.Path);
+
+                var b = beta.CreateInstance("Beta.B")!;
+                var writeMethod = b.GetType().GetMethod("Write")!;
+                var exception = Assert.Throws<TargetInvocationException>(
+                    () => writeMethod.Invoke(b, new object[] { sb, "Test B" }));
+                Assert.IsAssignableFrom<FileNotFoundException>(exception.InnerException);
+
+                var actual = sb.ToString();
+                Assert.Equal(@"", actual);
+            });
         }
 
         private static void VerifyAssemblies(IEnumerable<Assembly> assemblies, params (string simpleName, string version, string path)[] expected)
         {
             Assert.Equal(expected, Roslyn.Utilities.EnumerableExtensions.Order(assemblies.Select(assembly => (assembly.GetName().Name!, assembly.GetName().Version!.ToString(), assembly.Location))));
         }
 
-        [ConditionalFact(typeof(CoreClrOnly))]
+        /// <summary>
+        /// Verify the set of asesmblies loaded as analyzer dependencies are the specified assembly paths
+        /// </summary>
+        private static void VerifyDependencyAssemblies(DefaultAnalyzerAssemblyLoader loader, params string[] assemblyPaths)
+        {
+            IEnumerable<Assembly> loadedAssemblies;
+
+#if NETCOREAPP
+            // This verify only works where there is a single load context.
+            var alcs = DefaultAnalyzerAssemblyLoader.TestAccessor.GetOrderedLoadContexts(loader);
+            Assert.Equal(1, alcs.Length);
+
+            loadedAssemblies = alcs[0].Assemblies;
+#else
+
+            // Unfortunately in .NET Framework we cannot determine which Assemblies are in the LoadFrom context
```

Comment:
It looks like this comment should be revised, since we have a trick for determining if the assembly is in the LoadFrom context.